onnx/test/test_backend_test.py

```diff
@@ -41,8 +41,9 @@ def prepare(cls,
                 ):  # type: (...) -> Optional[onnx.backend.base.BackendRep]
         super(DummyBackend, cls).prepare(model, device, **kwargs)
 
-        # test shape inference
-        model = onnx.shape_inference.infer_shapes(model)
+        # test strict shape inference
+        model = onnx.shape_inference.infer_shapes(model, check_type=True, strict_mode=True)
+        onnx.checker.check_model(model, True)
```

Comment:
I suggest moving check_model above infer_shapes. No point in inferring shapes for an invalid model.